include/gpgmm_d3d12.h

```diff
@@ -210,12 +211,38 @@ namespace gpgmm::d3d12 {
     @param descriptor A reference to HEAP_DESC structure that describes the heap.
     @param pResidencyManager A pointer to the ResidencyManager used to manage this heap.
     @param createHeapFn  A callback function which creates a ID3D12Pageable derived type.
+    @param createHeapFnContext  A pointer to a class designed to implement the actual heap creation
+    function and store any necessary variables.
     @param[out] ppHeapOut Pointer to a memory block that receives a pointer to the
     heap.
+
+    Example call showing the usage of createHeapFn and createHeapFnContext:
+
+    CreateHeap(descriptor, pResidencyManager, CallbackContext:CallbackWrapper,
+    reinterpret_cast<void*>(callbackContext), ppHeapOut);
+
+    Example Callback Context Class:
+
+    class CallbackContext {
+        public:
+            CallbackContext(<Pass variables needed for heap creation here>);
+            CreateHeap(void *context, ID3D12Pageable** ppPageableOut);
+            static CallbackWrapper(ID3D12Pageable** ppPageableOut);
+        private:
+            <Declare variables needed for heap creation here>
+    }
+
+    Example Implementation of CallbackWrapper:
+
+    HRESULT CallbackContext:CallbackWrapper(void* context, ID3D12Pageable** ppPageableOut) {
+        CallbackContext* callbackContext = reinterpret_cast<CallbackContext*>(context);
+        return callbackContext->CreateHeap(ppPageableOut);
+    }
     */
     GPGMM_EXPORT HRESULT CreateHeap(const HEAP_DESC& descriptor,
```

Comment:
Alternatively, we could have a CreateHeap2 (and means to not include std::function).

Reply:
If I understand correctly, you mean to leave the old CreateHeap that can accept a standard callback function without a context class? Do you think that would be useful? You wouldn't be able to access any variables (i.e. the D3D device).

Reply:
Yea, basically. As you can probably tell, it's cumbersome to not use a `std::function` (STL-compatible) for this. Not only does it break the API, it's harder to use. So a `CreateHeap2` entry point (mainly just for Chromium) might unblock you faster.